### PR TITLE
[RLlib] auto API reference pages: `rllib/env` package only.

### PR DESCRIPTION
**Pull request by sven1977**
Auto API reference pages overhaul: `rllib/env` package only.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/package_ref/index.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+This section contains an overview of RLlib's package- and API reference.
+If you think there is anything missing, please open an issue on `Github`_.
+
+.. _`GitHub`: https://github.com/ray-project/ray/issues
+
+.. toctree::
+   :maxdepth: 2
+
+   env.rst
+   ../../rllib-package-ref.rst  .. TODO: Eventually, move all reference pages into this dir here (doc/source/rllib/package_ref/)
```
Comment (richardliaw) [inline doc/source/rllib/package_ref/index.rst:19]: i don't think the bottom rendered properly

Comment (sven1977) [reply doc/source/rllib/package_ref/index.rst:19]: Yeah, I'll check ...

Comment (sven1977) [reply doc/source/rllib/package_ref/index.rst:19]: done

File: doc/source/rllib/package_ref/index.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. include:: ../we_are_hiring.rst
+
+.. _rllib-reference-docs:
+
+RLlib API Reference
+===================
+
+.. tip:: We'd love to hear your feedback on using RLlib - `ask your questions or give feedback at <https://discuss.ray.io>`_!
```
Comment (richardliaw) [inline doc/source/rllib/package_ref/index.rst:8]: this part doesn't render properly btw

**Review by richardliaw — APPROVED** [review]

File: doc/source/rllib/package_ref/index.rst (hunk 2 above)
Comment (sven1977) [reply doc/source/rllib/package_ref/index.rst:8]: Looks actually ok on my laptop.

image

Comment (sven1977) [reply doc/source/rllib/package_ref/index.rst:8]: Great catch. href doesn't get rendered, of course. Fixed

**Pull request merged**